text/3478-better-spread-operator.md

```diff
@@ -0,0 +1,106 @@
+- Feature Name: better-spread-operator
+- Start Date: 2023-07-23
+- RFC PR: [rust-lang/rfcs#3478](https://github.com/rust-lang/rfcs/pull/3478)
+
+# Summary
+[summary]: #summary
+
+- Allow a value to be spread into another value of a different type
+- Allow partial field spreading
+
+# Motivation
+[motivation]: #motivation
+
+We currently have the spread operator, which allows a struct to be constructed with default values provided by other instance of the struct:
```

Comment:
In Rust, this is called functional update syntax: https://doc.rust-lang.org/stable/reference/expressions/struct-expr.html#functional-update-syntax

I personally don't think that name is good, but calling it spread isn't very good either: spreads in general are for splicing a list of items into another list, and the reason why it works in JavaScript is because objects are general maps and not actual structs. Just because Rust structs "feel" like objects doesn't mean they're the same thing. Here, saying `Struct { field: val, ..base }` means, genuinely "copy `base`, then set `field` to `val`" and not "set `field` to `val`, then set the rest of the fields from `base`", which is genuine distinction worth making semantically.